src/GapicClientTrait.php

```diff
@@ -250,12 +264,37 @@ private function buildClientOptions(array $options)
 
         // mTLS: If no apiEndpoint has been supplied by the user, and either
         // GOOGLE_API_USE_MTLS_ENDPOINT tells us to, or mTLS is available, use the mTLS endpoint.
-        if ($options['apiEndpoint'] === $defaultOptions['apiEndpoint']
-            && $this->shouldUseMtlsEndpoint($options)
-        ) {
-            $options['apiEndpoint'] = self::determineMtlsEndpoint($options['apiEndpoint']);
+        if (is_null($apiEndpoint) && $this->shouldUseMtlsEndpoint($options)) {
```

Comment:
**NOTE** - we previously checked if a user supplied the API endpoint by seeing if it was different from the default. Now we are checking to see if it's supplied in the options array. This requires consideration.

In this case, if a user supplied the endpoint directly, even if it was the same as the default, this would mean it should be used instead of the mTLS endpoint, so while this is a change of behavior, to me it seems like a bug fix.  However, if we want to be extra safe, we can revert this.